lib/postProcess.js

```diff
@@ -47,9 +47,14 @@ postProcess._fetchRelatedResources = function(request, mainResource, callback) {
 
   resourcesToFetch = Object.keys(resourcesToFetch).map(function(type) {
     var ids = resourcesToFetch[type];
-    var urlJoiner = "&filter[id]=";
-    ids = urlJoiner + ids.join(urlJoiner);
-    return jsonApi._apiConfig.pathPrefix + type + "/?" + ids;
+    var qsField = "filter[id]=";
+    var urlJoiner = "&" + qsField;
+    ids = qsField + ids.join(urlJoiner);
+    var uri = jsonApi._apiConfig.pathPrefix + type + "/?" + ids;
+    if (request.route.query) {
+      uri += "&" + request.route.query;
+    }
+    return uri;
```

Comment:
It wasn't just that, it was because we weren't passing the filter to the included resources rerouted request (see `54-56`). With the new strategy of not altering the original filter and adding the parsed filter as a new property we won't need to touch `lib/postProcess.js` at all so I am going to remove these changes.